qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -234,22 +239,20 @@ def __init__(self, name: str, address: str, terminator: str = "\n",
                            label='Voltage',
                            unit='V',
                            set_cmd=partial(self._get_set_output, "VOLT"),
-                           get_cmd=partial(self._get_set_output, "VOLT")
+                           get_cmd=partial(self._get_set_output, "VOLT"),
+                           snapshot_exclude=(True if self.source_mode() ==
+                           'CURR' else False)
                            )
-        # Again, at init we are in "VOLT" mode. Hence, exclude the snapshot of
-        # 'current' as instrument does not support this parameter in
-        # "VOLT" mode.
+
         self.add_parameter('current',
                            label='Current',
                            unit='I',
                            set_cmd=partial(self._get_set_output, "CURR"),
                            get_cmd=partial(self._get_set_output, "CURR"),
-                           snapshot_exclude=True
+                           snapshot_exclude=(True if self.source_mode() ==
+                           'VOLT' else False)
                            )
 
-        # This is changed through the source_mode interface
-        self.output_level = self.voltage
```

Comment:
You can't remove this. It is used within _ramp_source. Rather it needs to be set to voltage or current depending on the mode that we are in

Reply:
That's correct.

Reply:
@jenshnielsen 
Do you think it is a good idea to define the output_level as a delegate parameter like the range?

Reply:
Yes 👍